scss/components/_button.scss

```diff
@@ -158,14 +158,17 @@ $button-transition: background-color 0.25s ease-out, color 0.25s ease-out !defau
 }
 
 /// Adds disabled styles to a button by fading the element, reseting the cursor, and disabling pointer events.
-@mixin button-disabled($color: $primary-color) {
+@mixin button-disabled(
+  $background: $primary-color,
+  $color: $button-color
+) {
   opacity: $button-opacity-disabled;
   cursor: not-allowed;
+  background-color: $background;
+  
+  $color: color-pick-contrast($background, ($button-color, $button-color-alt));
```

Comment:
hmm... It should not. But two things:
- No need to write here whole part of codes.
- This is your PR, I can help you but cannot fix everything 😄 . Use `@debug` to show the values of your variables and try to understand the changes of `$color`.